app/models/async_transaction/vet360/base.rb

```diff
@@ -2,6 +2,70 @@
 
 module AsyncTransaction
   module Vet360
-    class Base < AsyncTransaction::Base; end
+    class Base < AsyncTransaction::Base
+
+      FINAL_STATUSES = [
+        'REJECTED',
+        'COMPLETED_SUCCESS',
+        'COMPLETED_NO_CHANGES_DETECTED',
+        'COMPLETED_FAILURE',
+        #'RECEIVED_DEAD_LETTER_QUEUE' #@TODO Do we know what this is?
+      ]
+
+      # Updates the status and transaction_status with fresh API data
+      # @params user [User] the user whose tx data is being updated
+      # @params service [Vet360::ContactInformation::Service] an initialized vet360 client
+      # @params tx_id [int] the transaction_id
+      # @returns [AsyncTransaction::Vet360::Base]
+      def self.refresh_transaction_status(user, service, tx_id = nil)
+
+        transaction_record = Base.find_by(user_uuid: user.uuid, transaction_id: tx_id)
+
+        raise Common::Exceptions::RecordNotFound, transaction_record unless transaction_record
+
+        # No need for a API request if this tx is already complete
+        return transaction_record if transaction_record.finished?
+
+        api_response = Base.fetch_transaction(user, transaction_record, service)
+
+        transaction_record.status = END_STATUS if FINAL_STATUSES.include? api_response.transaction.status
+        transaction_record.transaction_status = api_response.transaction.status
+        transaction_record.save!
+
+        return transaction_record
+
+      end
+
+      # Requests a transaction from vet360 for an app transaction
+      # @params user [User] the user whose tx data is being updated
+      # @params transaction_record [AsyncTransaction::Vet360::Base] the tx record to be checked
+      # @params service [Vet360::ContactInformation::Service] an initialized vet360 client
+      # @returns [Vet360::Models::Transaction]
+      def self.fetch_transaction(user, transaction_record, service)
+        
+        response = case transaction_record
```

Comment:
I don't believe this needs to be assigned, since its not being used later in this method.